In [2]:
import pandas as pd
import numpy as np
import json

Importing the data as a dataframe.

In [3]:
dataset = pd.read_csv("data/Data_Lv3_UMDOTS_Escooters.csv")

Lets take a look at a few data point

In [4]:
#dataset.head()
dataset.tail()

Unnamed: 0  Unnamed: 0.1                    CREATED  \
40320       40320         40320  2020-10-24 22:21:40+00:00   
40321       40321         40321  2020-10-24 21:28:43+00:00   
40322       40322         40322  2020-10-25 01:22:00+00:00   
40323       40323         40323  2020-10-25 00:55:24+00:00   
40324       40324         40324  2020-10-24 06:03:05+00:00   

                           START                        END  DISTANCE  \
40320  2020-10-24 22:21:40+00:00  2020-10-24 22:26:38+00:00     0.794   
40321  2020-10-24 21:28:43+00:00  2020-10-24 21:32:54+00:00     0.404   
40322  2020-10-25 01:22:00+00:00  2020-10-25 01:26:16+00:00     0.629   
40323  2020-10-25 00:55:24+00:00  2020-10-25 01:07:38+00:00     0.567   
40324  2020-10-24 06:03:05+00:00  2020-10-24 06:38:05+00:00     5.220   

       RIDE_ID  CUSTOMER_ID  MINUTES VEHICLE_TYPE  \
40320  3304114       322401        5      Scooter   
40321  3303630       298095        5      Scooter   
40322  3305589       604319        5      Scooter   
40323  3305336       610943       13      Scooter   
40324  3299791       324757       35      Scooter   

                                                    PATH  \
40320  {\n  "coordinates": [\n    [\n      -76.943936...   
40321  {\n  "coordinates": [\n    [\n      -76.935917...   
40322  {\n  "coordinates": [\n    [\n      -76.940231...   
40323  {\n  "coordinates": [\n    [\n      -76.944152...   
40324  {\n  "coordinates": [\n    [\n      -76.934166...   

                                              TIMESTAMPS  START LONG  \
40320  [\n  "2020-10-24T22:21:46Z",\n  "2020-10-24T22...  -76.943937   
40321  [\n  "2020-10-24T21:28:49Z",\n  "2020-10-24T21...  -76.935917   
40322  [\n  "2020-10-25T01:22:07Z",\n  "2020-10-25T01...  -76.940231   
40323  [\n  "2020-10-25T00:55:30Z",\n  "2020-10-25T00...  -76.944153   
40324  [\n  "2020-10-24T06:03:12Z",\n  "2020-10-24T06...  -76.934167   

       START LAT   END LONG    END LAT  
40320  38.988110 -76.934104  38.992061  
40321  38.986389 -76.933818  38.992196  
40322  38.996377 -76.945600  38.992842  
40323  38.991956 -76.940216  38.987199  
40324  38.990645 -76.936088  38.982452

In [5]:
dataset.shape

(40325, 16)

In [6]:
dataset.dtypes

Unnamed: 0        int64
Unnamed: 0.1      int64
CREATED          object
START            object
END              object
DISTANCE        float64
RIDE_ID           int64
CUSTOMER_ID       int64
MINUTES           int64
VEHICLE_TYPE     object
PATH             object
TIMESTAMPS       object
START LONG      float64
START LAT       float64
END LONG        float64
END LAT         float64
dtype: object

In [7]:
# All the data types do not match lets fix them

# Convert as Timestamp data
dataset["CREATED"] = pd.to_datetime(dataset["CREATED"])
# Convert as Timestamp data
dataset["START"] = pd.to_datetime(dataset["START"])
# Convert as Timestamp data
dataset["END"] = pd.to_datetime(dataset["END"])
# Convert timestamp as list
dataset["TIMESTAMPS"] = dataset["TIMESTAMPS"].apply(eval)

The dataset has 40325 rows and 16 columns

In [8]:
# Get the year of each data entry
# Initialize new columns
dataset["Year"] = dataset["CREATED"][0].year
for index, item in dataset.iterrows():
    dataset["Year"][index] = dataset["CREATED"][index].year

<ipython-input-8-71b62617ba7d>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dataset["Year"][index] = dataset["CREATED"][index].year


In [9]:
ride_count_year = dataset.groupby("Year").size().reset_index(name='ride_count')
ride_count_year

Year  ride_count
0  2019       18382
1  2020       21943

The distance column is missing values

In [10]:
dataset["DISTANCE"].isna().sum()
#dataset["DISTANCE"].dropna()

2273

Total of 2273 missing datapoint for distance. No sure how to deal with this yet.

Next we separate all ride by years: October 2020 and October 2019

In [12]:
dataset.groupby("CREATED")

In [13]:

dataset["START POINT"] = (dataset["START LAT"].astype(str)) + ", " + dataset["START LONG"].astype(str)

dataset["END POINT"] = (dataset["END LAT"].astype(str)) + ", " + dataset["END LONG"].astype(str)
dataset.head()

Unnamed: 0  Unnamed: 0.1                   CREATED  \
0           0             0 2020-10-03 03:32:15+00:00   
1           1             1 2020-10-03 03:07:39+00:00   
2           2             2 2020-10-03 03:17:05+00:00   
3           3             3 2020-10-03 02:24:58+00:00   
4           4             4 2020-10-03 03:19:34+00:00   

                      START                       END  DISTANCE  RIDE_ID  \
0 2020-10-03 03:32:15+00:00 2020-10-03 03:32:46+00:00     0.008  3135665   
1 2020-10-03 03:07:39+00:00 2020-10-03 03:37:06+00:00     1.062  3135395   
2 2020-10-03 03:17:05+00:00 2020-10-03 03:24:24+00:00     0.847  3135503   
3 2020-10-03 02:24:58+00:00 2020-10-03 03:11:56+00:00     3.447  3134955   
4 2020-10-03 03:19:34+00:00 2020-10-03 03:22:45+00:00     0.386  3135529   

   CUSTOMER_ID  MINUTES VEHICLE_TYPE  \
0       193964        1      Scooter   
1       542947       30      Scooter   
2       517321        8      Scooter   
3       578051       47      Scooter   
4       552305        4      Scooter   

                                                PATH  \
0  {\n  "coordinates": [\n    [\n      -76.934163...   
1  {\n  "coordinates": [\n    [\n      -76.938180...   
2  {\n  "coordinates": [\n    [\n      -76.937599...   
3  {\n  "coordinates": [\n    [\n      -76.943553...   
4  {\n  "coordinates": [\n    [\n      -76.938115...   

                                          TIMESTAMPS  START LONG  START LAT  \
0       [2020-10-03T03:32:14Z, 2020-10-03T03:32:30Z]  -76.934163  38.990656   
1  [2020-10-03T03:07:46Z, 2020-10-03T03:12:16Z, 2...  -76.938181  38.982087   
2  [2020-10-03T03:17:12Z, 2020-10-03T03:17:32Z, 2...  -76.937600  38.979613   
3  [2020-10-03T02:25:04Z, 2020-10-03T02:26:24Z, 2...  -76.943553  38.982470   
4  [2020-10-03T03:19:40Z, 2020-10-03T03:19:45Z, 2...  -76.938116  38.982084   

    END LONG    END LAT  Year                             START POINT  \
0 -76.934285  38.990739  2020                  38.9906555, -76.934163   
1 -76.949165  38.992849  2020  38.982087166666666, -76.93818083333333   
2 -76.933335  38.992676  2020   38.97961266666667, -76.93759983333334   
3 -76.943546  38.982476  2020  38.982470166666666, -76.94355316666666   
4 -76.943257  38.986101  2020          38.9820835, -76.93811566666666   

                                END POINT  
0  38.990738666666665, -76.93428483333334  
1   38.99284866666667, -76.94916483333333  
2           38.992676, -76.93333466666667  
3  38.982475666666666, -76.94354633333333  
4                   38.986101, -76.943257

2020    21943
2019    18382
Name: Year, dtype: int64